tools/launch_benchmark_suite_scheduler_slurm.py

```diff
@@ -0,0 +1,82 @@
+import sys
+
+import submitit
+from fvcore.common.file_io import PathManager
+from vissl.config.attr_dict import AttrDict
+from vissl.engines.benchmark_suite_scheduler import BenchmarkSuiteScheduler
+from vissl.utils.io import load_file
+
+
+# Default slurm options to pass to the executor.
+_DEFAULT_SLURM_OPTIONS = {
+    "NAME": "vissl",
+    "COMMENT": "vissl evaluation job",
+    "PARTITION": "learnlab",
+    "CONSTRAINT": "",
+    "TIMEOUT_MIN": 72 * 60,  # Timeout in minutes.
+    "CPUS_PER_TASK": 8,
+    "MEM_GB": 16,
+    "ADDITIONAL_PARAMETERS": {},
+}
+
+
+class SlurmEvaluatorJob:
+    def __init__(self, benchmark_suite_scheduler: BenchmarkSuiteScheduler):
+        self.benchmark_suite_scheduler = benchmark_suite_scheduler
+
+    def __call__(self):
+        self.benchmark_suite_scheduler.evaluate()
+
+    def checkpoint(self):
+        self.benchmark_suite_scheduler.save_evaluation_results()
+        self.benchmark_suite_scheduler.autoload_benchmark_suite_scheduler_checkpoint = (
+            True
+        )
+
+        trainer = SlurmEvaluatorJob(
+            benchmark_suite_scheduler=self.benchmark_suite_scheduler
+        )
+        return submitit.helpers.DelayedSubmission(trainer)
+
+
+def launch_benchmark_suite_scheduler(config_file):
+    assert PathManager.exists(config_file), "Slurm evaluator config file must exist."
+
+    config = load_file(config_file)
+    benchmark_suite_scheduler = BenchmarkSuiteScheduler(**config["params"])
+    benchmark_suite_scheduler_job = SlurmEvaluatorJob(
+        benchmark_suite_scheduler=benchmark_suite_scheduler
+    )
+    executor = submitit.AutoExecutor(folder=benchmark_suite_scheduler.evaluation_dir())
+
+    override_slurm_options = config.get("slurm_options", {})
+    slurm_options = {**_DEFAULT_SLURM_OPTIONS, **override_slurm_options}
+    slurm_options = AttrDict(slurm_options)
+
+    executor.update_parameters(
+        name=slurm_options.NAME,
+        slurm_comment=slurm_options.COMMENT,
+        slurm_partition=slurm_options.PARTITION,
+        slurm_constraint=slurm_options.CONSTRAINT,
+        timeout_min=slurm_options.TIMEOUT_MIN,
+        nodes=1,
+        cpus_per_task=slurm_options.CPUS_PER_TASK,
+        tasks_per_node=1,
```

Comment:
question: do we need to impose this constraint? aka can use configure this and override too?

Reply:
I think we should set this to 1. There will only be 1 task run here on SLURM. (If you set multiple tasks, it will run the same job multiple times with the same config).